docs/kap_proposals/kap_6_hashicorp_vault.md

```diff
@@ -0,0 +1,124 @@
+* VAULT_ADDR: URL for vault  
+* VAULT_SKIP_VERIFY=true: if set, do not verify presented TLS certificate before communicating with Vault server. Setting this variable is not recommended except during testing  
+* VAULT_TOKEN: token for vault or file (~/.vault-tokens)  
+* VAULT_ROLE_ID: (required by approle)  
+* VAULT_SECRET_ID: (required by approle)  
+* VAULT_USERNAME: username to login to vault  
+* VAULT_PASSWORD: password to login to vault  
+* VAULT_CLIENT_KEY: the path to an unencrypted PEM-encoded private key matching the client certificate  
+* VAULT_CLIENT_CERT: the path to a PEM-encoded client certificate for TLS authentication to the Vault server  
+* VAULT_CACERT: the path to a PEM-encoded CA cert file to use to verify the Vault server TLS certificate  
+* VAULT_CAPATH: the path to a directory of PEM-encoded CA cert files to verify the Vault server TLS certificate  
+* VAULT_NAMESPACE: specify the Vault Namespace, if you have one  
+  
+Considering a key-value pair like `my_key`:`my_secret` ( in our case let’s store `hello`:`batman` inside the vault ) in the path `secret/foo` in a kv-v2(KV version 2) secret engine on the vault server, to use this as a secret either follow:  
+  
+```shell  
+$ echo "foo:hello" > somefile.txt  
+$ kapitan secrets --write vaultkv:path/to/secret_inside_kapitan --file somefile.txt --target dev-sea  
+```  
+or in a single line  
+```shell  
+$ echo "foo:hello"  | kapitan secrets --write vaultkv:path/to/secret_inside_kapitan -t dev-sea -f -  
+```  
+The entire string __"foo:hello"__ is base64 encoded and stored in the secret_inside_kapitan. Now secret_inside_kapitan contains the following  
+  
+```yaml    
+data: Zm9vOmhlbGxvCg==  
+encoding: original  
+type: vaultkv  
+```  
+  
+Encoding tells the type of data given to kapitan, if it is `original` then after decoding base64 we'll get the original secret and if it is `base64` then after decoding once we still have a base64 encoded secret and have to decode again.  
+Parameters in the secret file are collected from the inventory of the target we gave from CLI `--target dev-sea`. If target isn't provided then kapitan will identify the variables from the environment, but providing `auth` is necessary as a key inside target parameters like the one shown:  
+```yaml  
+parameters:
+  kapitan:
+    secrets:
+      vaultkv:
+        auth: userpass
+        addr: http://127.0.0.1:8200
+        engine: kv-v2
+        namespace: CICD-alpha
+        mount: team-alpha-secret
+        skip_verify: false
+        client_key: /path/to/key
+        client_cert: /path/to/cert
+```
+Almost all the environment variables can be defined in kapitan inventory except the token, password, secret_id, etc. Environment like `VAULT_TOKEN`,`VAULT_USERNAME`,`VAULT_PASSWORD`,`VAULT_ROLE_ID`,` VAULT_SECRET_ID` 
```

Comment:
To avoid user confusion, can we rewrite this as a clearer note? Maybe split it to two sentences saying which envs can be declared in the kapitan inventory and which can't?